packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -283,7 +284,10 @@ function mountCustomElement(
 
     if (vm) {
         if (process.env.NODE_ENV !== 'production') {
-            assert.isTrue(vm.state === VMState.created, `${vm} cannot be recycled.`);
+            // FIXME: why is the vm.state "connected" sometimes?
+            if (!features.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+                assert.isTrue(vm.state === VMState.created, `${vm} cannot be recycled.`);
+            }
```

Comment:
Not sure why this happens sometimes. Disabling the assert doesn't seem to impact the behavior at all.